heron/lib/widgets/BookmarksPanel.js

```diff
@@ -557,11 +609,13 @@ Heron.widgets.BookmarksPanel = Ext.extend(Heron.widgets.HTMLPanel, {
 			name: this.scName,
 			desc: this.scDesc,
 			layers: this.scvisibleLayers,
+			layersopacity: this.scvisibleLayersOpacity,

```

Comment:
I think the bookmark format is a bit too flat. After all it becomes a JSON structure. This makes it hard to for future additions. For example: for WMS-dimension layers (timesliders etc) we may want to store current parameter settings. Also with catalog services (see Catalog example in http://kadviewer.kademo.nl) we may add Layers dynamically. At least reloading a bookmark should not fail as the Layer name may not be present.

Also the version is still '1'. I think there is an opportunity here to restructure the bookmark object to be more like a real OGC WMC: for example: the 'layers' could be an array of Layer objects, like : 

       layer: {
          name: 'naam',
          opacity: <number>,
          type: 
          edits:
          params: 
       }

Instead of adding a new Array for each Layer property. By the version number you may detect and migrate old bookmarks.

I realize that your current solution will work for your application, but there are already issues for dealing with dynamically added Layers. 

Another question: what happens with the Editing Layer Styles (or any other Vector Layer) when these are changed via the Vector Styler? This could be a challenge as well.